kafka-client/src/main/java/dev/responsive/kafka/internal/db/CassandraWindowedTable.java

```diff
@@ -98,9 +103,33 @@ public class CassandraWindowedTable implements
   // as these entities have different views of the current time and should not be unified.
   // (Specifically, this table will always lag the view of stream-time that is shared by the
   // ResponsiveWindowStore and CommitBuffer due to buffering/batching of writes)
-  private long lastFlushStreamTime = UNINITIALIZED_STREAM_TIME;
-  private long pendingFlushStreamTime = UNINITIALIZED_STREAM_TIME;
-  private SegmentRoll activeRoll;
+  private final Map<Integer, PendingFlushInfo> kafkaPartitionToPendingFlushInfo = new HashMap<>();
+
+  // TODO: move this into the LWTWriter/Factory to keep this class stateless
```

Comment:
I know this is super ugly, I intend to do this TODO right away (though probably in a separate followup PR)

Reply:
Ok actually I know I said I would do this immediately but I'm actually going to start on the Mongo implementation first, so I can see how it makes sense to fit together with that since we won't be using the LwtWriter for Mongo but will presumably still need the same time-tracking stuff

I'll do it as part of, or else immediately after, the Mongo work